### DISCUSSION TITLE
Sorting controls by argTypes order

### DISCUSSION THREAD

**Discussion by aldrichdev** [storybookjs/storybook#24524] [Ideas]
### Is your feature request related to a problem? Please describe.

I have the following `argTypes` in a `meta` object of my story file:
```
  argTypes: {
    chevronPosition: { name: 'Chevron Position', options: ['left', 'right'] },
    href: { name: 'Href', control: 'text' },
    children: { name: 'Text', control: 'text' },
    title: { name: 'Title', control: 'text' },
    target: { table: { disable: true } },
    styles: { table: { disable: true } },
    rel: { table: { disable: true } },
    id: { table: { disable: true } },
  },
```

However, this order is not reflected on the Storybook page under Controls (which is expected at this time).

![image](https://github.com/storybookjs/storybook/assets/17816529/c93d78d4-6e5f-4118-b8a7-dde85cebecdd)

The only ways to sort controls appear to be 1) alphabetically, 2) required first, or 3) not at all, none of which really work for me.

### Describe the solution you'd like

It would be nice if there was an option to have the order of controls in the Controls tab of the UI match the order they are listed in `argTypes` - perhaps a new `parameters.controls.sort` setting like `byArgTypes`.

### Describe alternatives you've considered

The existing sort options - `alpha`, `requiredFirst`, `none` - none of which are ideal

### Are you able to assist to bring the feature to reality?

yes, I can

**Comment by kornjiradej** [comment]
You can sort the order of controls by using `args`

For example:
`args: {
    chevronPosition: 'left',
    href: 'www.test.com',
    title: 'Try me!'
  }
`

> ### Is your feature request related to a problem? Please describe.
> I have the following `argTypes` in a `meta` object of my story file:
> 
> ```
>   argTypes: {
>     chevronPosition: { name: 'Chevron Position', options: ['left', 'right'] },
>     href: { name: 'Href', control: 'text' },
>     children: { name: 'Text', control: 'text' },
>     title: { name: 'Title', control: 'text' },
>     target: { table: { disable: true } },
>     styles: { table: { disable: true } },
>     rel: { table: { disable: true } },
>     id: { table: { disable: true } },
>   },
> ```
> 
> However, this order is not reflected on the Storybook page under Controls (which is expected at this time).
> 
> ![image](https://private-user-images.githubusercontent.com/17816529/276733800-c93d78d4-6e5f-4118-b8a7-dde85cebecdd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY5NTUwMDgsIm5iZiI6MTcxNjk1NDcwOCwicGF0aCI6Ii8xNzgxNjUyOS8yNzY3MzM4MDAtYzkzZDc4ZDQtNmU1Zi00MTE4LWI4YTctZGRlODVjZWJlY2RkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI5VDAzNTE0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEwNGRhOWEyM2Q0Y2VhYmNlNDk1YTZiMDNlMDVjZDg0NGQ3NjliZmVmMzBlODhjYWJiMDJhZmEyY2NjNzZlYjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qdSLLrlAqAvfxWYXL_J4rPzzSF8yza86xaJXVgshZ0s)
> 
> The only ways to sort controls appear to be 1) alphabetically, 2) required first, or 3) not at all, none of which really work for me.
> 
> ### Describe the solution you'd like
> It would be nice if there was an option to have the order of controls in the Controls tab of the UI match the order they are listed in `argTypes` - perhaps a new `parameters.controls.sort` setting like `byArgTypes`.
> 
> ### Describe alternatives you've considered
> The existing sort options - `alpha`, `requiredFirst`, `none` - none of which are ideal
> 
> ### Are you able to assist to bring the feature to reality?
> yes, I can
> 
> ### Additional context
> _No response_

**Reply by Shinigami92** [reply]
This does not satisfy my usecase where I have `argTypes` that are not covered by `args`

e.g. I have expose in a category and I want them after props category (talking about a Vue component)

The order should be props, events, slots, expose
And then by argType.name

**Reply by Shinigami92** [reply]
I opened a PR to address this
- #28565
